src/impl/realm_coordinator.cpp

```diff
@@ -81,7 +81,14 @@ void RealmCoordinator::create_sync_session()
     if (m_sync_session)
         return;
 
-    m_sync_session = SyncManager::shared().get_session(m_config.path, *m_config.sync_config);
+    util::Optional<std::array<char, 64>> encryption_key;
+    if (m_config.encryption_key.data()) {
+        std::array<char, 64> key;
+        std::copy_n(m_config.encryption_key.begin(), key.size(), key.begin());
+        encryption_key = std::move(key);
+    }
+
+    m_sync_session = SyncManager::shared().get_session(m_config.path, *m_config.sync_config, std::move(encryption_key));
```

Comment:
std::move() on a std::array doesn't do anything useful (the data for std::array is stored inline and not in a separate stack allocation).